### PR TITLE
docs(google-maps): Improve documentation for Google Maps.

### PR DESCRIPTION
**Pull request by mbehrlich**
Add new Google Maps components to documentation, including MapPolygon,
MapPolyline, MapRectangle, MapCircle, MapGroundOverlay, MapKmlLayer,
MapTransitLayer, MapTrafficLayer, and MapBicyclingLayer. Add
instructions on how to lazy load the Google Maps JavaScript API for use
with the @angular/google-maps components.

### CODE REVIEW

File: src/google-maps/README.md [hunk 1]
```diff
@@ -14,109 +14,119 @@ To install, run `npm install @angular/google-maps`.
...
+- [`MapTransitLayer`](./map-transit-layer/README.md)
+- [`MapBicyclingLayer`](./map-bicycling-layer/README.md)
 
-  removeLastMarker() {
-    this.markerPositions.pop();
-  }
-}
+## The Options Input
+
+The Google Maps components implement all of the options for their respective objects from the Google Maps JavaScript API through an `options` input, but they also have specific inputs for some of the most common options. For example, the google maps component could have its options set either in with a google.maps.MapOptions object:
```
Comment (jelbourn) [inline src/google-maps/README.md:107]:
```suggestion
The Google Maps components implement all of the options for their respective objects from the Google Maps JavaScript API through an `options` input, but they also have specific inputs for some of the most common options. For example, the Google Maps component could have its options set either in with a google.maps.MapOptions object:
```

File: src/google-maps/README.md [hunk 2]
```diff
@@ -14,109 +14,119 @@ To install, run `npm install @angular/google-maps`.
...
-</google-map>
-
-<div>Latitude: {{display?.lat}}</div>
-<div>Longitude: {{display?.lng}}</div>
+```typescript
+center: google.maps.LatLngLiteral = {lat: 40, lng: -20};
+zoom = 4;
 ```
+
+Not every option has its own input. See the api for each component to see if the option has a dedicated input or if it should be set in the options input.
```
Comment (jelbourn) [inline src/google-maps/README.md:132]:
```suggestion
Not every option has its own input. See the API for each component to see if the option has a dedicated input or if it should be set in the options input.
```

**Review by jelbourn — APPROVED** [review]
LGTM, just a couple tiny capitalization nits

**Comment by weilinzung** [comment]
@mbehrlich 

The lazyload example has an issue with `You have included the Google Maps JavaScript API multiple times on this page. This may cause unexpected errors.` when the same Google map component is used multiple times within a single-page app.

**Pull request merged**